Dalamud/Game/Gui/ContextMenu/ContextMenuAddressResolver.cs

```diff
@@ -0,0 +1,38 @@
+namespace Dalamud.Game.Gui.ContextMenu;
+
+/// <summary>
+/// <see cref="ContextMenuGui"/> memory address resolver.
+/// </summary>
+internal class ContextMenuAddressResolver : BaseAddressResolver
+{
+    /// <summary>
+    /// Gets the address of the RaptureAtkModule.OpenAddonByAgent member function.
+    /// This is called whenever an agent requests an addon to be opened.
+    /// We only check if addons ContextMenu or AddonContextSub are requested and mark which agent is opening the addon.
+    /// Only AgentInventoryContext and AgentContext open these two.
+    /// </summary>
+    public nint ContextAddonOpenByAgent { get; private set; }
+    
+    /// <summary>
+    /// Gets the address of the AddonContextMenu.vf72 member function.
+    /// This is called whenever a menu item is clicked on.
+    /// </summary>
+    public nint ContextMenuOnVf72 { get; private set; }
+
+    /// <summary>
+    /// Gets the address of the RaptureAtkModule.OpenAddon member function.
+    /// We call this whenever we need to open a submenu.
+    /// </summary>
+    public nint RaptureAtkModuleOpenAddon { get; private set; }
+
+    /// <summary>
+    /// Scan for and setup any configured address pointers.
+    /// </summary>
+    /// <param name="sig">The signature scanner to facilitate setup.</param>
+    protected override void Setup64Bit(ISigScanner sig)
+    {
+        this.ContextAddonOpenByAgent = sig.ScanText("E8 ?? ?? ?? ?? 0F B7 C0 48 83 C4 60");
```

Comment:
Are these signatures not present in ClientStructs? If not, is there a compelling reason to have them here?

Reply:
The intent of putting them in CS is to reduce the maintenance burden during patch cycles - instead of having to run sig checks and update processes across multiple projects and systems, it's all in one place. 

Ideally we'd be working on minimizing our sig influence to structures and systems that don't exist or can't easily exist in CS in its current state.